include/ginkgo/core/matrix/dense.hpp

```diff
@@ -1267,30 +1291,33 @@ struct temporary_clone_helper<matrix::Dense<ValueType>> {
 /**
  * Creates a view of a given Dense vector.
  *
- * @tparam ValueType  the underlying value type of the vector
+ * @tparam VecPtr  a (smart or raw) pointer to the vector.
  *
  * @param vector  the vector on which to create the view
  */
-template <typename ValueType>
-std::unique_ptr<matrix::Dense<ValueType>> make_dense_view(
-    matrix::Dense<ValueType>* vector)
+template <typename VecPtr>
+std::unique_ptr<matrix::Dense<typename detail::pointee<VecPtr>::value_type>>
+make_dense_view(VecPtr&& vector)
 {
-    return matrix::Dense<ValueType>::create_view_of(vector);
+    using value_type = typename detail::pointee<VecPtr>::value_type;
+    return matrix::Dense<value_type>::create_view_of(vector);
 }
 
 
 /**
  * Creates a view of a given Dense vector.
  *
- * @tparam ValueType  the underlying value type of the vector
+ * @tparam VecPtr  a (smart or raw) pointer to the vector.
  *
  * @param vector  the vector on which to create the view
  */
-template <typename ValueType>
-std::unique_ptr<const matrix::Dense<ValueType>> make_const_dense_view(
-    const matrix::Dense<ValueType>* vector)
+template <typename VecPtr>
+std::unique_ptr<
+    const matrix::Dense<typename detail::pointee<VecPtr>::value_type>>
+make_const_dense_view(VecPtr&& vector)
```

Comment:
side note: Because create_const_view_of works only for Dense, it should be fine.
Otherwise, it can accept any type with value_type